<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EditorTimelineWidget</class>
 <widget class="QWidget" name="EditorTimelineWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>886</width>
    <height>44</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>EditorTimelineWidget</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <property name="spacing">
    <number>1</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="timelineSpeedLayout">
     <property name="spacing">
      <number>1</number>
     </property>
     <property name="leftMargin">
      <number>7</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>10</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QSlider" name="m_SpeedSlider">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>60</width>
         <height>15</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>60</width>
         <height>15</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>30</number>
       </property>
       <property name="value">
        <number>10</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="m_SpeedSpinBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>25</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="layoutDirection">
        <enum>Qt::RightToLeft</enum>
       </property>
       <property name="frame">
        <bool>true</bool>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::NoButtons</enum>
       </property>
       <property name="decimals">
        <number>1</number>
       </property>
       <property name="minimum">
        <double>0.100000000000000</double>
       </property>
       <property name="maximum">
        <double>5.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="horizontalSpacer_2">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>10</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QPushButton" name="m_PlayStopButton">
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="m_PrevFrameButton">
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="m_NextFrameButton">
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="m_SkipBackwardButton">
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="m_SkipForwardButton">
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="m_DirectionButton">
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="EditorTimelineSlider" name="m_TimelineSlider">
     <property name="styleSheet">
      <string notr="true">QSlider::groove:horizontal {
border: 1px solid #bbb;
background: white;
height: 4px;
}

QSlider::sub-page:horizontal {
/*background: qlineargradient(x1: 0, y1: 0,    x2: 0, y2: 1,
    stop: 0 #808080, stop: 1 #808080);
background: qlineargradient(x1: 0, y1: 0.2, x2: 1, y2: 1,
    stop: 0 #808080, stop: 1 #808080);*/
border: 1px solid #777;
height: 10px;
}

QSlider::add-page:horizontal {
background: #fff;
border: 1px solid #777;
height: 10px;
}

QSlider::handle:horizontal {
background: qlineargradient(x1:0, y1:0, x2:1, y2:1,
    stop:0 #eee, stop:1 #ccc);
border: 1px solid #777;
width: 10px;
margin-top: -9px;
margin-bottom: -9px;
margin-left: -1px;
}

QSlider::handle:horizontal:hover {
background: qlineargradient(x1:0, y1:0, x2:1, y2:1,
    stop:0 #fff, stop:1 #ddd);
border: 1px solid #444;
}

QSlider::sub-page:horizontal:disabled {
background: #bbb;
border-color: #999;
}

QSlider::add-page:horizontal:disabled {
background: #eee;
border-color: #999;
}

QSlider::handle:horizontal:disabled {
background: #eee;
border: 1px solid #aaa;
}</string>
     </property>
     <property name="sliderPosition">
      <number>0</number>
     </property>
     <property name="tracking">
      <bool>true</bool>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="invertedAppearance">
      <bool>false</bool>
     </property>
     <property name="invertedControls">
      <bool>false</bool>
     </property>
     <property name="tickPosition">
      <enum>QSlider::NoTicks</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="m_FrameInfoLable">
     <property name="minimumSize">
      <size>
       <width>40</width>
       <height>0</height>
      </size>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>EditorTimelineSlider</class>
   <extends>QSlider</extends>
   <header>EditorTimelineSlider.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>m_PlayStopButton</sender>
   <signal>clicked()</signal>
   <receiver>EditorTimelineWidget</receiver>
   <slot>tooglePlay()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>15</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_DirectionButton</sender>
   <signal>clicked()</signal>
   <receiver>EditorTimelineWidget</receiver>
   <slot>toogleDirection()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>175</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_NextFrameButton</sender>
   <signal>clicked()</signal>
   <receiver>EditorTimelineWidget</receiver>
   <slot>nextFrame()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>79</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_PrevFrameButton</sender>
   <signal>clicked()</signal>
   <receiver>EditorTimelineWidget</receiver>
   <slot>prevFrame()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>47</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_SkipForwardButton</sender>
   <signal>clicked()</signal>
   <receiver>EditorTimelineWidget</receiver>
   <slot>skipForward()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>143</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_SkipBackwardButton</sender>
   <signal>clicked()</signal>
   <receiver>EditorTimelineWidget</receiver>
   <slot>skipBackward()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>111</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_SpeedSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>EditorTimelineWidget</receiver>
   <slot>speedSliderChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>225</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_SpeedSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>EditorTimelineWidget</receiver>
   <slot>speedSpinBoxChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>269</x>
     <y>19</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>tooglePlay()</slot>
  <slot>toogleDirection()</slot>
  <slot>nextFrame()</slot>
  <slot>prevFrame()</slot>
  <slot>skipForward()</slot>
  <slot>skipBackward()</slot>
  <slot>speedSpinBoxChanged(double)</slot>
  <slot>speedSliderChanged(int)</slot>
  <slot>timeSliderChanged(int)</slot>
  <slot>slot1()</slot>
 </slots>
</ui>
